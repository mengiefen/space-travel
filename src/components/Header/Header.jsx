import React from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar, Container, Nav } from 'react-bootstrap';
import './Header.scss';

const Header = () => {
  const NavItems = [
    {
      link: '/',
      name: 'Rocket',
    },
    {
      link: 'missions',
      name: 'Missions',
    },
    {
      link: 'my-profile',
      name: 'My Profile',
    },
  ];
  return (
    <>
      <Navbar bg="dark" variant="dark">
        <Container>
          <Navbar.Brand href="#home">Navbar</Navbar.Brand>
          <Nav className="me-auto">
            {NavItems.map((navItem) => (
              <NavLink
                to={navItem.link}
                key={navItem.link}
                className={({ isActive }) => {
                  if (isActive) return 'nav-link active';
                  return 'nav-link';
                }}
              >
                {navItem.name}
              </NavLink>
            ))}
          </Nav>
        </Container>
      </Navbar>
    </>
  );
};

export default Header;
