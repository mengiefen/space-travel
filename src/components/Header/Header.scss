.active {
  color: #fff;
  text-decoration: underline;
}

.logo-img {
  max-width: 3rem;
  margin: 0 1rem;
}
