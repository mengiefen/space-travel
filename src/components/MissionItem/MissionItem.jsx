import { PropTypes } from 'prop-types';
import Button from 'react-bootstrap/Button';
import './MissionItem.scss';

const MissionItem = (props) => {
  const {
    id, name, description, reserved, handleJoinMission, handleLeaveMission,
  } = props;

  return (
    <tr id={id}>
      <td className="td-name">{name}</td>
      <td className="td-description">{description}</td>
      <td className="td-badge">
        {reserved ? (
          <Button variant="primary" className="btn-reserved" disabled>ACTIVE MEMBER</Button>
        ) : (
          <Button variant="secondary" className="btn-reserved">NOT A MEMBER</Button>
        )}
      </td>
      <td className="td-button">
        {reserved ? (
          <Button
            type="button"
            variant="secondary"
            onClick={() => handleLeaveMission(id)}
          >
            Leave Mission
          </Button>
        ) : (
          <Button
            type="button"
            variant="outline-secondary"
            onClick={() => handleJoinMission(id)}
          >
            Join Mission
          </Button>
        )}
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
  handleJoinMission: PropTypes.func.isRequired,
  handleLeaveMission: PropTypes.func.isRequired,
};

export default MissionItem;
