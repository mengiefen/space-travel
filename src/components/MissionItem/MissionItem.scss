.missions-container {
  border: 1px solid grey;
  border-radius: 15px;
  width: 100%;
  display: flex;
  justify-content: center;
  margin: auto;
  margin-bottom: 4rem;
  padding: 1rem;
  background-color: rgba(128, 128, 128, 0.2);
  overflow: hidden;
}

.td-name {
  width: 10%;
  font-weight: bold;
}

.td-description {
  width: 70%;
}

.td-badge {
  width: 10%;
  text-align: center;
}

span {
  font-size: 1rem;
  font-weight: bold;
  color: white;
  border-radius: 8px;
  padding: 1rem;
  background-color: grey;
}

.td-button {
  width: 10%;
  text-align: center;

  button {
    font-size: 1.5rem;
    background-color: transparent;
    cursor: pointer;
    border: 1px solid gray;
    border-radius: 5px;
    color: gray;
    transition: color linear 0.2s, border-color;
    padding: 0.6rem;

    &:hover {
      color: blueviolet;
      border-color: blueviolet;
    }

    &:active {
      color: blueviolet;
      box-shadow: inset 0 0 5px blueviolet;
    }
  }
}
