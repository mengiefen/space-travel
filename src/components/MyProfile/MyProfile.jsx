import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';

const MyProfile = () => {
  const [reservedRockets, setReservedRockets] = useState([]);
  const [reservedMissions, setReservedMissions] = useState([]);
  const rockets = useSelector((state) => state.rockets);
  const missions = useSelector((state) => state.missionsReducer);
  useEffect(() => {
    setReservedRockets(rockets.filter((rocket) => rocket.reserved === true));
    setReservedMissions(missions.filter((mission) => mission.reserved === true));
  });

  return (
    <div className="list">
      <ListGroup className="reserved-list">
        <h2 className="rocket-title">My Rockets</h2>
        {reservedRockets.map((rocket) => (
          <ListGroup.Item key={rocket.id} disabled>
            {rocket.name}
          </ListGroup.Item>
        ))}
      </ListGroup>
      <ListGroup className="reserved-list">
        <h2 className="rocket-title">My Missions</h2>
        {reservedMissions.map((mission) => (
          <ListGroup.Item key={mission.id} disabled>
            {mission.name}
          </ListGroup.Item>
        ))}
      </ListGroup>
    </div>
  );
};

export default MyProfile;
