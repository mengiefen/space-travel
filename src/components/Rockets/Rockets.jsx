import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import {
  cancelReservedRocket,
  getAllRockets,
  reserveRocket,
} from '../../redux/rockets/rockets';
import 'bootstrap/dist/css/bootstrap.min.css';
import './Rockets.scss';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getAllRockets());
  }, []);
  return (
    <>
      {rockets.map((rocket) => (
        <Card className="rockets-card" key={rocket.id}>
          <Card.Img
            variant="top"
            src={rocket.rocket_img}
            className="rockets-img"
          />
          <Card.Body className="rockets-detail">
            <Card.Title>{rocket.name}</Card.Title>
            <Card.Text>
              {rocket.reserved && (
                <Button variant="primary" className="btn-reserved">
                  Reserved
                </Button>
              )}
              {rocket.description}
            </Card.Text>
            <Button
              variant={rocket.reserved ? 'outline-secondary' : 'secondary'}
              onClick={() => {
                if (rocket.reserved) dispatch(cancelReservedRocket(rocket.id));
                else dispatch(reserveRocket(rocket.id));
              }}
            >
              {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rockets'}
            </Button>
          </Card.Body>
        </Card>
      ))}
    </>
  );
};

export default Rockets;
