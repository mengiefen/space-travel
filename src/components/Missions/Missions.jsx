import React from 'react';
import { useSelector } from 'react-redux';
import MissionItem from '../MissionItem/MissionItem';
import '../MissionItem/MissionItem.scss';

const Missions = () => {
  const missionList = useSelector((state) => state.missionsReducer);

  return (
    <section className="missions-container">
      <table striped bordered hover>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Content</th>
          </tr>
        </thead>
        <tbody>
          {
            missionList.map((mission) => (
              <MissionItem
                key={mission.id}
                id={mission.id}
                name={mission.name}
                description={mission.description}
              />
            ))
          }
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
