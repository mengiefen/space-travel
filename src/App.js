import { Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from './redux/configureStore';
import Header from './components/Header/Header';
import Rockets from './components/Rockets/Rockets';
import Missions from './components/Missions/Missions';
import MyProfile from './components/MyProfile/MyProfile';

function App() {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Header />
        <Routes>
          <Route path="/" element={<Rockets />} />
          <Route path="missions" element={<Missions />} />
          <Route path="my-profile" element={<MyProfile />} />
        </Routes>
      </PersistGate>
    </Provider>
  );
}

export default App;
