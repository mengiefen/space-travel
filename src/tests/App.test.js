import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../redux/configureStore';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MyProfile from '../components/MyProfile/MyProfile';
import Rockets from '../components/Rockets/Rockets';
import Missions from '../components/Missions/Missions';
import MissionItem from '../components/MissionItem/MissionItem';
import Header from '../components/Header/Header';
import App from '../App'

it('MyProfile renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Provider store={store}>
    <MyProfile />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Rockets renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Provider store={store}>
    <Rockets />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Missions renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Provider store={store}>
    <Missions />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('MissionItem renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Provider store={store}>
    <MissionItem />
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Header renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Router>
    <Header />
  </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('App renders correctly when there are no items', () => {
  const tree = renderer.create(
  <Router>
    <App />
  </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});