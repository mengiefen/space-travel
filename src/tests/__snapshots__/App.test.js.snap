// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly when there are no items 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt="logo"
        className="logo-img"
        src="logo.png"
      />
      Space Traveler's Hub
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Rocket
      </a>
      <a
        className="nav-link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className="nav-link"
        href="/my-profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </div>
  </div>
</nav>
`;

exports[`Header renders correctly when there are no items 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt="logo"
        className="logo-img"
        src="logo.png"
      />
      Space Traveler's Hub
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Rocket
      </a>
      <a
        className="nav-link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className="nav-link"
        href="/my-profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </div>
  </div>
</nav>
`;

exports[`MissionItem renders correctly when there are no items 1`] = `
<tr>
  <td
    className="td-name"
  />
  <td
    className="td-description"
  />
  <td
    className="td-badge"
  >
    <button
      className="btn-reserved btn btn-secondary"
      disabled={false}
      type="button"
    >
      NOT A MEMBER
    </button>
  </td>
  <td
    className="td-button"
  >
    <button
      className="btn btn-outline-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;

exports[`Missions renders correctly when there are no items 1`] = `
<div
  className="missions-container container"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Content
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`MyProfile renders correctly when there are no items 1`] = `
<div
  className="list"
>
  <div
    className="reserved-list list-group"
    onKeyDown={[Function]}
  >
    <h2
      className="rocket-title"
    >
      My Rockets
    </h2>
  </div>
  <div
    className="reserved-list list-group"
    onKeyDown={[Function]}
  >
    <h2
      className="rocket-title"
    >
      My Missions
    </h2>
  </div>
</div>
`;

exports[`Rockets renders correctly when there are no items 1`] = `null`;
